### PR TITLE
feature/skip decompile step if local version matches the already decompiled version

### PR DESCRIPTION
**Pull request by B13rg**
#83 

compares the entire file, so if any of the entries change it'll go through the decompile process.
also added a --force flag in case one want to decompile anyway

### CODE REVIEW

**Comment by B13rg** [comment]
Example output:
```
❯ docker-compose -f ./docker-compose-local.yml up test1                                                   ─╯
[+] Building 0.0s (0/0)                                                                       docker:default
[+] Running 1/0
 ✔ Container deadbot-test1-1  Created                                                                   0.0s 
Attaching to test1-1
test1-1  | Decompiling source files...
test1-1  | version.txt and steam.inf have the same content.
test1-1  | ! Skipping Parser !
test1-1  | ! Skipping Changelogs !
test1-1  | ! Skipping DeadBot !
test1-1  | 
test1-1  | Done!
test1-1 exited with code 0
```

**Comment by Surxe** [comment]
@B13rg is this ready to be reviewed? If so, please request review from at least Saag :)

File: docker-compose-testing.yml
Comment (hassayag) [inline docker-compose-testing.yml:1]: Is this just for manual testing?

**Review by hassayag — CHANGES_REQUESTED** [review]
👌

File: docker-compose-testing.yml
Comment (B13rg) [reply docker-compose-testing.yml:1]: Sort of?  It's how I've been testing the env vars, but it's not exactly very useful on it's own....
I'll just remove it!

File: src/decompiler/decompile.py [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
     # Define paths
     os.makedirs(WORK_DIR, exist_ok=True)
-    os.system(f'cp "{DEADLOCK_PATH}/game/citadel/steam.inf" "{WORK_DIR}/version.txt"')
-    os.system(f'cp "{DEADLOCK_PATH}/game/citadel/steam.inf" "{OUTPUT_DIR}/version.txt"')
+    steam_inf_path = f'{DEADLOCK_PATH}/game/citadel/steam.inf'
+    version_path = f'{WORK_DIR}/version.txt'
+
+    # if the version files match, nothing to do
+    if filecmp.cmp(steam_inf_path, version_path):
+        print('version.txt and steam.inf have the same content.')
```
[the remark this answers is not in the document]
Comment (B13rg) [reply src/decompiler/decompile.py]: Done!
I added a helper func in the utils folder to parse the config file into a dict

File: src/decompiler/decompile.py [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
 
-def decompile(DEADLOCK_PATH, WORK_DIR, OUTPUT_DIR, DECOMPILER_CMD):
+def decompile(DEADLOCK_PATH, WORK_DIR, DECOMPILER_CMD, force=False):
+    """
+    Decompiles deadlock game files and generated parsed output.
+
+    Args:
+        DEADLOCK_PATH (str): The path to the deadlock game files.
+        WORK_DIR (str): The working directory for the decompilation process.
+        DECOMPILER_CMD (str): The command used to run the decompiler.
```
[the remark this answers is not in the document]
Comment (B13rg) [reply src/decompiler/decompile.py:15]: Done!  Also updated readme with flag

**Comment by B13rg** [comment]
Should be all ready to go!

**Review by hassayag — APPROVED** [review]

**Pull request merged**